crates/testing-framework/src/ephemeral/mod.rs

```diff
@@ -0,0 +1,248 @@
+
+    pub fn add_peer_with_neighborhood(
+        &mut self,
+        peer: Peer,
+        neighborhood: impl IntoIterator<Item = impl Into<PeerId>>,
+    ) -> &mut PeerWithNeighborhood {
+        let peer_id = peer.peer_id.clone();
+        let mut peer_with_neigh = PeerWithNeighborhood::new(peer);
+        peer_with_neigh.extend_neighborhood(neighborhood.into_iter().map(Into::into));
+        self.peers
+            .insert(peer_id.clone(), Rc::new(peer_with_neigh.into()));
+        Rc::get_mut(self.peers.get_mut(&peer_id).unwrap())
+            .unwrap()
+            .get_mut()
+    }
+
+    /// Add a peer with default neighborhood.
+    pub fn add_peer(&mut self, peer: Peer) -> &mut PeerWithNeighborhood {
+        let peer_id = peer.peer_id.clone();
+        let mut peer_with_neigh = PeerWithNeighborhood::new(peer);
+        peer_with_neigh.extend_neighborhood(self.default_neiborhood.iter().cloned());
+        self.peers
+            .insert(peer_id.clone(), Rc::new(peer_with_neigh.into()));
+        Rc::get_mut(self.peers.get_mut(&peer_id).unwrap())
+            .unwrap()
+            .get_mut()
+    }
+
+    pub fn set_peer_failed<Id>(&mut self, peer_id: &Id, failed: bool)
+    where
+        PeerId: Borrow<Id>,
+        Id: Hash + Eq + ?Sized,
+    {
+        self.peers
+            .get_mut(peer_id)
+            .expect("unknown peer")
+            .as_ref()
+            .borrow_mut()
+            .set_failed(failed);
+    }
+
+    pub fn fail_peer_for<Id>(&mut self, source_peer_id: &Id, target_peer_id: impl Into<PeerId>)
+    where
+        PeerId: Borrow<Id>,
+        Id: Hash + Eq + ?Sized,
+    {
+        self.peers
+            .get_mut(source_peer_id)
+            .expect("unknown peer")
+            .as_ref()
+            .borrow_mut()
+            .get_neighborhood_mut()
+            .fail(target_peer_id);
+    }
+
+    pub fn unfail_peer_for<Id1, Id2>(&mut self, source_peer_id: &Id1, target_peer_id: &Id2)
+    where
+        PeerId: Borrow<Id1>,
+        Id1: Hash + Eq + ?Sized,
+        PeerId: Borrow<Id2>,
+        Id2: Hash + Eq + ?Sized,
+    {
+        self.peers
+            .get_mut(source_peer_id)
+            .expect("unknown peer")
+            .as_ref()
+            .borrow_mut()
+            .get_neighborhood_mut()
+            .unfail(target_peer_id);
+    }
+
+    // TODO there is some kind of unsymmetry between these methods and the fail/unfail:
+    // the latters panic on unknown peer; perhaps, it's OK
+    pub fn get_peer_env<Id>(&self, peer_id: &Id) -> Option<Rc<RefCell<PeerWithNeighborhood>>>
+    where
+        PeerId: Borrow<Id>,
+        Id: Hash + Eq + ?Sized,
+    {
+        self.peers.get(peer_id).cloned()
+    }
+
+    pub fn iter_execution<'s, Id>(
+        &'s self,
+        air: &'s str,
+        peer_id: &Id,
+    ) -> Option<impl Iterator<Item = RawAVMOutcome> + 's>
+    where
+        PeerId: Borrow<Id>,
+        Id: Eq + Hash + ?Sized,
+    {
+        let peer = self.get_peer_env(peer_id);
```

Comment:
This line confuses me. I expect that `get_peer_env` returns some kind of environment, but not a peer. `get_peer_with_env` will be much more understandable here.

Reply:
I've just renamed `PeerWithNeighborhood` to `PeerEnv` and made variable names consistent.